File: Graph(그래프)/백준_11724_연결 요소의 개수/연결 요소의 개수.cpp
#include <iostream>
#include <vector>
using namespace std;

bool check[1001];
vector<int> arr_list[1001];

void DFS(int n)
{
	for (int i = 0; i < arr_list[n].size(); i++)
	{
		int v = arr_list[n][i];

		if (check[v] == false)
		{
			check[v] = true;
			DFS(v);
		}
	}
}

int main()
{
	int N = 0, M = 0;
	scanf("%d %d", &N, &M);
	int count = 0;

	while (M--)
	{
		int a = 0, b = 0;
		scanf("%d %d", &a, &b);
		arr_list[a].push_back(b);
		arr_list[b].push_back(a);
	}

	for (int i = 1; i <= N; i++)
	{
		if (check[i] == false)
		{
			check[i] = true;
			count++;
			DFS(i);
		}
	}

	printf("%d", count);
}

File: Graph(그래프)/백준_11724_연결 요소의 개수/연결요소의 개수.kt
import java.util.*
import kotlin.math.*

var check = BooleanArray(1001, {false})
var arr_list = Array<ArrayList<Int>>(1001) {ArrayList<Int>()}

fun DFS(n: Int){
    for(i in arr_list[n]){
        if(check[i] == false){
            check[i] = true
            DFS(i)
        }
    }
}

fun main(args: Array<String>){
    val s = Scanner(System.`in`)
    var N = s.nextInt()
    var M = s.nextInt()

    while(M-- >0){
        var a = s.nextInt()
        var b = s.nextInt()
        arr_list[a].add(b)
        arr_list[b].add(a)
    }

    var count = 0;
    for(i in 1..N){
        if(check[i] == false){
            check[i] = true
            count++
            DFS(i)
        }
    }
    print(count)
}